Purchases/Public/Purchases+async.swift

```diff
@@ -0,0 +1,271 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Purchases+async.swift
+//
+//  Created by Andrés Boedo on 24/11/21.
+
+import Foundation
+import StoreKit
+
+/// This extension holds the biolerplate logic to convert methods with completion blocks into async / await syntax.
+extension Purchases {
+
+    @available(iOS 15.0, macOS 12, tvOS 15.0, watchOS 8.0, *)
+    func logInAsync(_ appUserID: String) async throws -> (CustomerInfo, Bool) {
```

Comment:
A few comments here:

~~- Maybe I'm missing something for them not being public, but if they are, we could copy the docs here?~~
~~- Do we need the word `async` in these? I think overloading them calling them with the same name `logIn` would be simpler.~~

(I realized why this is separate, I like it)

- I was also going to suggest documenting the type of `Error` that this can throw (you can find many tweets of me asking the Swift core team about typed throws 😛)
- I'd suggest naming the return tuple values: `(info: CustomerInfo, created: Bool)`

Reply:
❤️ it. 

> I was also going to suggest documenting the type of Error that this can throw (you can find many tweets of me asking the Swift core team about typed throws 😛)

Definitely worth doing, but maybe as a separate PR, since we should audit which Errors are thrown by each method and also update the matching non-async declarations

> I'd suggest naming the return tuple values: (info: CustomerInfo, created: Bool)

Will do!